argus/backends/azure/client.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2017 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import abc
+import random
+import six
+
+from argus import util
+from azure.common.credentials import UserPassCredentials
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.storage import StorageManagementClient
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.compute import ComputeManagementClient
+
+PROVISIONING_STATE_CREATING = 'Creating'
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseAzureClient(object):
+    """Class for managing Azure instances
+
+    """
+    def __init__(self, subscription_id=None, username=None, password=None,
+                 resource_group_name=None, storage_account_name=None,
+                 availability_zone=None, image_vhd_path=None, vm_username=None,
+                 vm_password=None):
+        self.vm_disk_size = 50000
+        self._subscription_id = subscription_id
+        self._username = username
+        self._password = password
+        self._resource_group_name = resource_group_name
+        self._storage_account_name = storage_account_name
+        self._availability_zone = availability_zone
+        self._image_vhd_path = image_vhd_path
+        self._vm_username = vm_username
+        self.vm_password = vm_password
+        self._creds = UserPassCredentials(username, password)
+        self.res_client = ResourceManagementClient(
+            self._creds, subscription_id)
```

Comment:
Just a suggestion, couldn't we save this to a variable since we are using it to initialize other objects too?